analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -167,6 +167,16 @@ def perform_analysis(args, skip_handler, context, actions, metadata_tool,
     Additionally, insert statistical information into the metadata dict.
     """
 
+    # FIXME: Ensure all mutable data structures are wrapped in manager proxy
+    #        objects before passing the data structure to other processes via
+    #        map_async.
+    #        Note that (only) top-level immutability is not sufficient.
```

Comment:
Did you found any description about this?
For the config_map we could use the frozen version you mentioned because they should not be changed.

Reply:
We have tried the deep-copy approach, however it failed to remove the buggy behaviour. This could still mean that something else is also improperly shared, but for now I'll just reorganize the comments, and leave the dict-sharing method as is.

Reply:
Also I have found no indication as to why it is wrong to simply shared non-wrapped data structures if they are read-only. The documentation indicates that changes are not reflected in the shared data structure (https://docs.python.org/2/library/multiprocessing.html#sharing-state-between-processes), mentions that locking could be important as well. Still, the semantics of read-only state sharing is murky. From experience, the whole 3.8.x branch seems to produce the undesired behaviour.